File: wrappers/javascript/anoncreds-shared/src/api/CredentialRequestMetadata.ts
import type { JsonObject } from '../types'

import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export class CredentialRequestMetadata extends AnoncredsObject {
  public static fromJson(json: JsonObject) {
    return new CredentialRequestMetadata(
      anoncreds.credentialRequestMetadataFromJson({ json: JSON.stringify(json) }).handle
    )
  }
}

File: wrappers/dotnet/lib/Requests/CredentialRequestMetadata.cs
using AnonCredsNet.Models;

namespace AnonCredsNet.Requests;

public class CredentialRequestMetadata : AnonCredsObject
{
    internal CredentialRequestMetadata(long handle)
        : base(handle) { }

    internal static CredentialRequestMetadata FromJson(string json) =>
        FromJson<CredentialRequestMetadata>(json);
}
